bindings/generate_bind_cxx_extras.cpp

```diff
@@ -81,8 +81,51 @@
            COMPARE_TYPES(unsigned short, CTYPE)     \
            COMPARE_TYPES(unsigned int, CTYPE)       \
            COMPARE_TYPES(unsigned long, CTYPE)      \
+           COMPARE_TYPES(unsigned long long, CTYPE)
+
+/* Note: Mirrors SHMEM_BIND_CXX_COLL_AND_OR_XOR */
+#define NEED_COLL_AND_OR_XOR_ASSOC(CTYPE)           \
+           COMPARE_TYPES_FIRST(unsigned char, CTYPE)\
+           COMPARE_TYPES(short, CTYPE)              \
+           COMPARE_TYPES(unsigned short, CTYPE)     \
+           COMPARE_TYPES(int, CTYPE)                \
+           COMPARE_TYPES(unsigned int, CTYPE)       \
+           COMPARE_TYPES(long, CTYPE)               \
+           COMPARE_TYPES(unsigned long, CTYPE)      \
+           COMPARE_TYPES(long long, CTYPE)          \
+           COMPARE_TYPES(unsigned long long, CTYPE) 
+
+/* Note: Mirrors SHMEM_BIND_CXX_COLL_MIN_MAX */
+#define NEED_COLL_MIN_MAX_ASSOC(CTYPE)              \
+           COMPARE_TYPES_FIRST(short, CTYPE)        \
+           COMPARE_TYPES(unsigned short, CTYPE)     \
+           COMPARE_TYPES(unsigned char, CTYPE)      \
+           COMPARE_TYPES(int, CTYPE)                \
+           COMPARE_TYPES(int8_t, CTYPE)             \
+           COMPARE_TYPES(unsigned int, CTYPE)       \
+           COMPARE_TYPES(long, CTYPE)               \
+           COMPARE_TYPES(unsigned long, CTYPE)      \
+           COMPARE_TYPES(long long, CTYPE)          \
            COMPARE_TYPES(unsigned long long, CTYPE) \
+           COMPARE_TYPES(float, CTYPE)              \
+           COMPARE_TYPES(double, CTYPE)             \
+           COMPARE_TYPES(long double, CTYPE)
 
+/* Note: Mirrors SHMEM_BIND_CXX_COLL_SUM_PROD */
+#define NEED_COLL_SUM_PROD_ASSOC(CTYPE)             \
+           COMPARE_TYPES_FIRST(short, CTYPE)        \
+           COMPARE_TYPES(unsigned short, CTYPE)     \
+           COMPARE_TYPES(unsigned char, CTYPE)      \
+           COMPARE_TYPES(int, CTYPE)                \
+           COMPARE_TYPES(int8_t, CTYPE)             \
```

Comment:
Is `int8_t` a special case?

Reply:
Yeah.. I was getting a build error without it, a conflict with signed char I believe..